src/registrar/admin.py

```diff
@@ -716,15 +718,51 @@ def response_change(self, request, obj):
         return super().response_change(request, obj)
 
     def do_delete_domain(self, request, obj):
+        if not isinstance(obj, Domain):
+            # Could be problematic if the type is similar,
+            # but not the same (same field/func names) so we err out.
+            # We do not want to accidentally delete records.
+            raise ValueError("Object is not of type Domain")
         try:
-            obj.deleted()
+            obj.deletedInEpp()
             obj.save()
-        except Exception as err:
-            self.message_user(request, err, messages.ERROR)
+        except RegistryError as err:
+            # Human-readable mappings of ErrorCodes. Can be expanded.
+            error_messages = {
+                ErrorCode.OBJECT_STATUS_PROHIBITS_OPERATION:
+                    f"Cannot delete Domain when in status {obj.status}",
+                ErrorCode.OBJECT_ASSOCIATION_PROHIBITS_OPERATION:
+                    "This subdomain is being used as a hostname on another domain"
+            }
+
+            message = "Cannot connect to the registry"
+            if not err.is_connection_error():
+                # If nothing is found, will default to returned err
+                message = error_messages.get(err.code, err)
+            self.message_user(
+                request,
+                f"Error deleting this Domain: {message}",
+                messages.ERROR
+            )
+        except TransitionNotAllowed:
+            if obj.state == Domain.State.DELETED:
+                self.message_user(
+                    request,
+                    "This domain is already deleted",
+                    messages.INFO,
+                )
+            else:
+                self.message_user(
+                    request,
+                    "Error deleting this Domain: "
+                    f"Can't switch from state '{obj.state}' to 'deleted'"
+                    ,
+                    messages.ERROR,
+                )
         else:
```

Comment:
@zandercymatics I really like this error handling and custom messaging.  Do you also need to catch Exception since deletedInEpp could raise a generic Exception?

Reply:
Thats a fair point. It probably would be helpful to display that error message on the page. How much information do you think should be displayed to the end user? The entire error message or just a generic one ("An unspecified error occurred")?

Reply:
I do not know.  At some point, we will need to norm on the messaging provided to users.  For now, "An unspecified error occurred" seems fine to me.  But I think this case just needs to be handled 'somehow'.  Unless the generic Exception is handled already somewhere else and I am just not seeing it.